front/components/app/blocks/Input.tsx

```diff
@@ -40,12 +46,43 @@ export default function Input({
   onBlockDown: () => void;
   onBlockNew: (blockType: BlockType | "map_reduce" | "while_end") => void;
 }>) {
-  const handleSetDataset = (dataset: string) => {
+  const [isDatasetModalOpen, setIsDatasetModalOpen] = useState(false);
+  const { dataset, isDatasetLoading, isDatasetError, mutateDataset } =
+    useDataset(owner, app, block.config.dataset, true);
+
+  const handleSetDataset = async (datasetName: string) => {
     const b = shallowBlockClone(block);
-    b.config.dataset = dataset;
+    b.config.dataset = datasetName;
     onBlockUpdate(b);
   };
 
+  const onUpdate = _.debounce(
```

Comment:
As per IRL I think this may overwrite the state of your data set if the debounce + write + get races with a user edit? To be confirmed

Proposal as per IRL is to have a save/cancel (+ confirm in case of exsit by click) on the modal.